lib/modules/manager/nix/extract.spec.ts

```diff
@@ -0,0 +1,68 @@
+import { GitRefsDatasource } from '../../datasource/git-refs';
+import { id as nixpkgsVersioning } from '../../versioning/nixpkgs';
+import { extractPackageFile } from '.';
+
+describe('modules/manager/nix/extract', () => {
+  it('returns null when no nixpkgs', () => {
+    const content = `{
+  inputs = {};
+}`;
+    const res = extractPackageFile(content);
+
+    expect(res).toBeNull();
+  });
+
+  it('returns nixpkgs', () => {
+    const content = `{
+  inputs = {
+    nixpkgs.url = "github:nixos/nixpkgs/nixos-21.11";
+  };
+}`;
+
+    const res = extractPackageFile(content);
+
+    expect(res?.deps).toHaveLength(1);
+    expect(res?.deps).toEqual([
+      {
+        depName: 'nixpkgs',
+        currentValue: 'nixos-21.11',
+        datasource: GitRefsDatasource.id,
+        packageName: 'https://github.com/NixOS/nixpkgs',
+        versioning: nixpkgsVersioning,
+      },
+    ]);
+  });
+
+  it('is case insensitive', () => {
+    const content = `{
+  inputs = {
+    nixpkgs.url = "github:NixOS/nixpkgs/nixos-21.11";
+  };
+}`;
+
+    const res = extractPackageFile(content);
+
+    expect(res?.deps).toHaveLength(1);
+    expect(res?.deps).toEqual([
+      {
+        depName: 'nixpkgs',
+        currentValue: 'nixos-21.11',
+        datasource: GitRefsDatasource.id,
+        packageName: 'https://github.com/NixOS/nixpkgs',
+        versioning: nixpkgsVersioning,
+      },
+    ]);
+  });
+
+  it('ignores nixpkgs with no explicit ref', () => {
+    const content = `{
+  inputs = {
+    nixpkgs.url = "github:NixOS/nixpkgs";
```

Comment:
we probably should support this later, otherwise lockfile maintenance won't be possible without any found dependency.

Reply:
No explicit reference here means the default (`master`) branch. This supported, but much less used as these packages are not as well covered by the nix binary cache[^1] built by Hydra[^2].

Here's a quick estimate of how often implicit vs explicit references are used:

- [`path:flake.nix "\"github:nixos/nixpkgs\""` 588 results][1]
- [`path:flake.nix "\"github:nixos/nixpkgs/"` 3.8k results][2]

[^1]: https://nixos.wiki/wiki/Binary_Cache
[^2]: https://hydra.nixos.org/

[1]: https://github.com/search?q=path%3Aflake.nix%20%22%5C%22github%3Anixos%2Fnixpkgs%5C%22%22&type=code
[2]: https://github.com/search?q=path%3Aflake.nix+%22%5C%22github%3Anixos%2Fnixpkgs%2F%22&type=code